contracts/utils/structs/Checkpoints.sol

```diff
@@ -26,6 +26,11 @@ library Checkpoints {
         uint224 _value;
     }
 
+    /**
+     * @dev A value was attempted to be inserted on a past checkpoint.
+     */
+    error Checkpoint224PastInsert();
```

Comment:
> What do you think about CheckpointUnorderedInsertion?

I feel it's less expressive because order can be either before or after, whereas `Past` clearly defines a sequence. I'd keep `Past` even though this is for non-time.

Reply:
Solved as agreed with `CheckpointUnorderedInsertion `